### PR TITLE
Bump adler to a 1.0 version

### PR DESCRIPTION
**Pull request by Xanewok**
There should be no perceivable differences from this crate's PoV and 1.0 version inspires more confidence :smile: 

Differences: https://github.com/jonas-schievink/adler/compare/v0.2.3...v1.0.1

### CODE REVIEW

**Comment by oyvindln** [comment]
Could the breaking change in adler cause any issue for someone using both this and the adler crate? I doubt that's even a thing but just to make sure.

**Comment by Xanewok** [comment]
From the changelog:
### Breaking Changes

- `adler32_reader` now takes its generic argument by value instead of as a `&mut`.
- Renamed `adler32_reader` to `adler32`.

These are technically breaking changes but should not impact performance or safety, moreover it looks like it's not used at all by this crate so should be harmless to do and qualifies for only a patch-level semver bump :+1:

**Comment by oyvindln** [comment]
Yeah for miniz_oxide itself the change is not an issue, I meant if some crate that depended on both ```miniz_oxide``` and ```adler``` and used the reader would break, or would cargo handle using 2 different versions of it (provided both are specified in crates.toml?)

**Comment by Xanewok** [comment]
The latter - since 0.2.3 and 1.0 are *incompatible* in terms of semver, both have to be compiled and pulled by Cargo:
- 1.0 to be used by miniz_oxide 0.4.4 (hypothetical version that includes this PR)
- 0.2.3 if someone depends on both miniz_oxide 0.4.4 and adler 0.2.3.

If someone depends on both miniz_oxide 0.4.4 and adler 1.0, only one version of adler will be compiled and used.

tl;dr: nothing breaks, the only downside is potentially having to compile two versions of the adler crate if someone depends on both miniz_oxide and adler 0.2 *specifically*.
